src/logging/Types.h

```diff
@@ -0,0 +1,98 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+// Header for types shared between cluster and logging components.
+//
+// Currently these are in detail, but over time may move into the
+// public namespace once established.
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "zeek/IntrusivePtr.h"
+#include "zeek/threading/SerialTypes.h"
+
+namespace zeek {
+class EnumVal;
+using EnumValPtr = IntrusivePtr<EnumVal>;
+
+namespace logging::detail {
+
+/**
+ * A single log record.
+ *
+ * This is what a Zeek record value passed into Log::write()
+ * is converted into before passed to a local log writer or
+ * send via the cluster to a remote node.
+ */
+using LogRecord = std::vector<threading::Value>;
+
+/**
+ * A struct holding all necessary information that relates to
+ * log writes for a given path. These values are constant over
+ * the lifetime of a \a WriterFrontend.
+ */
+struct LogWriteHeader {
```

Comment:
What's the intuition for "Header" in the name?

Reply:
Ha, good one :-)

When a cluster node sends out a batch of log records, the message starts with a description of the batch to allow stateless interpretation of the batch (stream, writer, path, schema/fields). That's the "Header". And the "LogWrite" is more like "LogBatch".

Maybe "LogBatchHeader", let me know what you think, I'm a bit stuck in a box here :-)

Reply:
Thanks! Believe me when I say that I always at least hesitate when I bring up naming stuff. :-)

I think what tripped me up was that the members don't really have anything to do with a header, and I see now that it's just context that the serializer then turns into a "header" write. So perhaps something like `LogWriteInfo`. But, no biggie.

Reply:
I kept this one for now, but certainly up for post-merge-rename!